File: templates/Uwp/Projects/Default/Activation/DefaultActivationHandler.cs
using System;
using System.Threading.Tasks;
using Windows.ApplicationModel.Activation;

using Param_RootNamespace.Services;

namespace Param_RootNamespace.Activation
{
    internal class DefaultActivationHandler : ActivationHandler<IActivatedEventArgs>
    {
        protected override async Task HandleInternalAsync(IActivatedEventArgs args)
        {
            // When the navigation stack isn't restored, navigate to the first page and configure
            // the new page by passing required information in the navigation parameter
            object arguments = null;
            if (args is LaunchActivatedEventArgs launchArgs)
            {
                arguments = launchArgs.Arguments;
            }

            NavigationService.Navigate(_navElement, arguments);
            await Task.CompletedTask;
        }

        protected override bool CanHandleInternal(IActivatedEventArgs args)
        {
            // None of the ActivationHandlers has handled the app activation
            return NavigationService.Frame.Content == null && _navElement != null;
        }
    }
}

File: code/TemplateStudioForUWP/Templates/Uwp/Projects/Default._VB/Activation/DefaultActivationHandler.vb
Imports Param_RootNamespace.Services

Namespace Activation
    Friend Class DefaultActivationHandler
        Inherits ActivationHandler(Of IActivatedEventArgs)

        Protected Overrides Async Function HandleInternalAsync(args As IActivatedEventArgs) As Task
            ' When the navigation stack isn't restored, navigate to the first page and configure
            ' the new page by passing required information in the navigation parameter
            Dim arguments As Object = Nothing
            Dim launchArgs = TryCast(args, LaunchActivatedEventArgs)
            If launchArgs IsNot Nothing Then
                arguments = launchArgs.Arguments
            End If

            NavigationService.Navigate(_navElement, arguments)
            Await Task.CompletedTask
        End Function

        Protected Overrides Function CanHandleInternal(args As IActivatedEventArgs) As Boolean
            ' None of the ActivationHandlers has handled the app activation
            Return NavigationService.Frame.Content Is Nothing
        End Function
    End Class
End Namespace
